src/controllers/UserNewsController.js

```diff
@@ -0,0 +1,133 @@
+const axios = require("axios");
+const userNewsPreferences = require("../data/userNewsPreferences.json");
+const userReadNews = require("../data/userReadNews.json");
+const userFavouriteNews = require("../data/userFavouriteNews.json");
+const users = require("../data/users.json");
+const { updateUserNewsPreferences } = require("../helpers");
+const Validation = require("../utilities/Validation");
+
+const parsedUserPreferences = JSON.parse(JSON.stringify(userNewsPreferences));
+
+const fetchNews = (req, res) => {
+  const userId = req.id;
+  const userPreference = parsedUserPreferences[userId];
+  let url = `https://newsapi.org/v2/top-headlines?apiKey=${process.env.NEWSAPI_KEY}`;
+
+  if (userPreference) {
+    const promises = [];
+    Object.keys(userPreference).forEach((key) => {
+      if (userPreference[key] != null) {
+        promises.push(axios.get(`${url}&${key}=${userPreference[key]}`));
+      }
+    });
+    if (promises.length) {
+      Promise.all(promises)
+        .then((newsRes) => {
+          let news = [];
+          newsRes.forEach((newsData) => {
+            news = [...news, ...newsData.data.articles];
+          });
+          return res.status(200).json({
+            status: true,
+            message: "All news have been fetched successfully.",
+            data: news,
+          });
+        })
+        .catch((err) => {
+          return res.status(500).json({
+            status: false,
+            message: "Error while fetching news.",
+            data: null,
+          });
+        });
+    } else {
+      return res.status(400).json({
```

Comment:
can't we club these two error conditions together by checking if user preference is not empty, because a non-empty user preference list will anyway make promises list as non-empty